test/compilable/staticforeach.d

```diff
@@ -638,3 +638,36 @@ auto bug17660(){
     return x;
 }
 static assert(bug17660()==3);
+
+int breakContinueBan(){
+    static assert(!is(typeof({
+        for(;;){
+            static foreach(i;0..1){
+                break;
+            }
+        }
+    })));
+    static assert(!is(typeof({
+        for(;;){
+            static foreach(i;0..1){
+                continue;
+            }
+        }
+    })));
```

Comment:
Putting them in `fail_compilation/` allows for the exact error message generated to be tested, and so makes for a better test. There are plenty of examples in `fail_compilation/` to show how this is done - it's pretty easy.